packages/vuetify/src/components/VDataTable/VDataTable.sass

```diff
@@ -52,6 +52,9 @@
                 & .v-data-table-header__content
                   display: contents
 
+              &.v-data-table-column--hidden
+                display: none
```

Comment:
I am not totally sure if this needs to have `!important` here as the `d-none` utility has? But it worked fine in my tests.